newIDE/app/src/EventsSheet/InstructionEditor/InstructionOrObjectSelector.js

```diff
@@ -173,17 +173,44 @@ export default class InstructionOrObjectSelector extends React.PureComponent<
       this._scrollView.current.scrollTo(this._selectedItem.current);
     }
     const { allObjectsList, allGroupsList } = enumerateObjectsAndGroups(
-      this.props.globalObjectsContainer,
-      this.props.objectsContainer
+      this.props.projectScopedContainersAccessor
+        .get()
+        .getObjectsContainersList()
+    );
+
+    const objectsContainersList = this.props.projectScopedContainersAccessor
+      .get()
+      .getObjectsContainersList();
+
+    if (objectsContainersList.getObjectsContainersCount() === 0) {
+      throw new Error(
+        'Called enumerateObjectsAndGroups without any object container.'
+      );
+    }
+    // TODO Use a loop instead of looking for 2 object containers.
+    if (objectsContainersList.getObjectsContainersCount() > 2) {
+      console.error(
+        'Called enumerateObjectsAndGroups with more than 2 object containers.'
```

Comment:
Careful, these logs are actually misleading the future developer:

```suggestion
        'Used InstructionOrObjectSelector without any object container.'
      );
    }
    // TODO Use a loop instead of looking for 2 object containers.
    if (objectsContainersList.getObjectsContainersCount() > 2) {
      console.error(
        'Used InstructionOrObjectSelector with more than 2 object containers.'
```